--- libreoffice-6/solenv/bin/assemble-flatpak.sh~	2018-04-03 15:23:51.803596867 +0000
+++ libreoffice-6/solenv/bin/assemble-flatpak.sh	2018-04-04 14:03:30.887072971 +0000
@@ -37,6 +37,34 @@
   "$(dirname /app/share/icons/hicolor/"${i#"${PREFIXDIR?}"/share/icons/hicolor/}")"/org.libreoffice.LibreOffice-"${i##*/apps/libreoffice-}"
 done
 
+mkdir -p /app/share/runtime/locale
+for i in $(ls /app/libreoffice/program/resource)
+do
+  lang="${i%[_@]*}"
+  mkdir -p /app/share/runtime/locale/"${lang}"/resource
+  mv /app/libreoffice/program/resource/"${i}" /app/share/runtime/locale/"${lang}"/resource
+  ln -s ../../../share/runtime/locale/"${lang}"/resource/"${i}" /app/libreoffice/program/resource
+done
+
+for i in /app/libreoffice/share/registry/Langpack-*.xcd /app/libreoffice/share/registry/res/*.xcd
+do
+  basename="$(basename "${i}" .xcd)"
+  lang="${basename#Langpack-}"
+  lang="${lang#fcfg_langpack_}"
+  lang="${lang#registry_}"
+
+  # ship the base app with at least one Langpack/fcfg_langpack
+  if [ "${lang}" = "en-US" ]
+  then
+    continue
+  fi
+
+  lang="${lang%-*}"
+  mkdir -p /app/share/runtime/locale/"${lang}"/registry
+  mv "${i}" /app/share/runtime/locale/"${lang}"/registry
+  ln -rs /app/share/runtime/locale/"${lang}"/registry/"${basename}".xcd "${i}"
+done
+
 ## org.libreoffice.LibreOffice.appdata.xml is manually derived from the various
 ## inst/share/appdata/libreoffice-*.appdata.xml (at least recent GNOME Software
 ## doesn't show more than five screenshots anyway, so restrict to one each from
