--- libreoffice-1/solenv/bin/assemble-flatpak.sh~	2018-05-15 11:33:39.466673280 +0000
+++ libreoffice-1/solenv/bin/assemble-flatpak.sh	2018-05-15 12:04:51.321521517 +0000
@@ -20,15 +20,15 @@
 for i in "${PREFIXDIR?}"/share/applications/libreoffice-*.desktop
 do
  sed -e 's,^Exec=libreoffice,Exec=/app/libreoffice/program/soffice,' \
-  -e 's/^Icon=libreoffice-/Icon=org.libreoffice.LibreOffice-/' "$i" \
-  >/app/share/applications/org.libreoffice.LibreOffice-"${i#"${PREFIXDIR?}"/share/applications/libreoffice-}"
+  -e 's/^Icon=libreoffice-/Icon=org.libreoffice.LibreOffice./' "$i" \
+  >/app/share/applications/org.libreoffice.LibreOffice."${i#"${PREFIXDIR?}"/share/applications/libreoffice-}"
 done
-mv /app/share/applications/org.libreoffice.LibreOffice-startcenter.desktop \
+mv /app/share/applications/org.libreoffice.LibreOffice.startcenter.desktop \
  /app/share/applications/org.libreoffice.LibreOffice.desktop
 
 # Flatpak .desktop exports take precedence over system ones due to
 # the order of XDG_DATA_DIRS - re-associating text/plain seems a bit much
-sed -i "s/text\/plain;//" /app/share/applications/org.libreoffice.LibreOffice-writer.desktop
+sed -i "s/text\/plain;//" /app/share/applications/org.libreoffice.LibreOffice.writer.desktop
 
 ## icons/hicolor/*/apps/libreoffice-* ->
 ## icons/hicolor/*/apps/org.libreoffice.LibreOffice-*:
@@ -40,7 +40,7 @@
  cp -a "$i" \
   "$(dirname /app/share/icons/hicolor/"${i#"${PREFIXDIR?}"/share/icons/hicolor/}")"/"$(basename "$i")"
  cp -a "$i" \
-  "$(dirname /app/share/icons/hicolor/"${i#"${PREFIXDIR?}"/share/icons/hicolor/}")"/org.libreoffice.LibreOffice-"${i##*/apps/libreoffice-}"
+  "$(dirname /app/share/icons/hicolor/"${i#"${PREFIXDIR?}"/share/icons/hicolor/}")"/org.libreoffice.LibreOffice."${i##*/apps/libreoffice-}"
 done
 
 mkdir -p /app/share/runtime/locale
@@ -142,7 +142,7 @@
 # append the appdata for the different components
 for i in "${PREFIXDIR?}"/share/appdata/libreoffice-*.appdata.xml
 do
-  sed "1 d; s/<id>libreoffice/<id>org.libreoffice.LibreOffice/" "$i" \
+  sed "1 d; s/<id>libreoffice-/<id>org.libreoffice.LibreOffice./" "$i" \
     >>/app/share/appdata/org.libreoffice.LibreOffice.appdata.xml
 done
 
